src/writing-policies/go/04-valitation.md

```diff
@@ -0,0 +1,351 @@
+	}
+
+	// NOTE 3
+	data := gjson.GetBytes(
+		payload,
+		"request.object.metadata.labels")
+
+	// NOTE 4
+	data.ForEach(func(key, value gjson.Result) bool {
+		label := key.String()
+
+		// NOTE 5
+		if settings.DeniedLabels.Contains(label) {
+			err = fmt.Errorf("Label %s is on the deny list", label)
+			// stop iterating over labels
+			return false
+		}
+
+		// NOTE 6
+		regExp, found := settings.ConstrainedLabels[label]
+		if found {
+			// This is a constrained label
+			if !regExp.Match([]byte(value.String())) {
+				err = fmt.Errorf("The value of %s doesn't pass user-defined constraint", label)
+				// stop iterating over labels
+				return false
+			}
+		}
+
+		return true
+	})
+
+	// NOTE 7
+	if err != nil {
+		return kubewarden.RejectRequest(
+			kubewarden.Message(err.Error()),
+			kubewarden.NoCode)
+	}
+
+	return kubewarden.AcceptRequest()
+```
+
+The code has some `NOTE` section inside of it. Let's get through them:
+
+1. The function ensures the JSON payload is properly formatted. This
+  is done using a function provided by the `gjson` library
+2. The `Settings` instance is created using one of the constructor
+  methods we defined inside of `settings.go`
+3. We use a `gjson` selector to get the `label` map provided by the object
+  embedded into the request
+4. We use a `gjson` helper to iterate over the results of the query. If the query
+  has no results the loop will never take place.
+5. We look for the `label` of the object inside of the list of denied labels
+  provided by the user via the policy settings. If the `label` is a denied
+  one, we set the value of the `err` variable and exit from the loop (that
+  happens by returning `false` instead of `true`).
+6. We look for the `label` of the object inside of the list of constrained labels
+  provided by the user via the policy settings. When we have a match we use
+  the regular expression provided by the user to validate the value of the label.
+  If the validation fails, we set the value of the `err` variable and exit
+  from the loop (that happens by returning `false` instead of `true`).
+7. If the `err` variable is not `nil`, we use the helper provided by Kubewarden's
+  SDK to reject the request. Otherwise we approve it.
+
+## Testing the validation code
+
+It's now time to write some unit tests to ensure the validation code is behaving
+properly. These tests are going to be located inside of the `validate_test.go`
+file.
+
+The tests will rely on some [test fixtures](https://en.wikipedia.org/wiki/Test_fixture#Software)
+located inside of the `test_data` directory. This directory has already
+been populated by the template repository.
+
+We will start by including the following packages:
+
+```go
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/deckarep/golang-set"
+	kubewarden_testing "github.com/kubewarden/policy-sdk-go/testing"
+)
+```
+
+The first unit test will ensure that having no user settings leads
+to the request to be approved:
```

Comment:
Should we use the same terminology? `Accepted` and in other instances of approve later on.

Reply:
I fixed that across of all the file (also inside of the code)